mmdet/models/dense_heads/decoupled_solo_head.py

```diff
@@ -0,0 +1,505 @@
+import numpy as np
+import mmcv
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import ConvModule, bias_init_with_prob, normal_init
+from mmdet.core import multi_apply, bbox2roi, matrix_nms
+from ..builder import HEADS, build_loss
+from .base_dense_seg_head import BaseDenseSegHead
+
+INF = 1e8
+
+from scipy import ndimage
+
+def points_nms(heat, kernel=2):
+    # kernel must be 2
+    hmax = nn.functional.max_pool2d(
+        heat, (kernel, kernel), stride=1, padding=1)
+    keep = (hmax[:, :, :-1, :-1] == heat).float()
+    return heat * keep
+
+def dice_loss(input, target):
```

Comment:
`dice_loss` is duplicated in `decoupled_solo_head` and `solo_head`. It can be moved into `mmdet.losses` as a `DiceLoss` module to reduce duplicated codes.